### ISSUE TITLE
Apps menu when multiple servers configured

### ISSUE THREAD

**Issue by r2evans** [nextcloud/desktop#1748]
### Expected behaviour

Apps menu (https://github.com/nextcloud/desktop/pull/195) allows choosing which (when multiple) server to use.

### Actual behaviour

It arbitrarily chooses a server.

Just now, when it first appeared, it showed all apps available on my second (as listed) server. A moment later, the menu of apps available changed to those available on my first listed server, and now that's all I can see and access via the menu.

### Steps to reproduce
1. Configure multiple (different) nextcloud instances
2. Access the "Apps" sub-menu from the tray right-click menu, noting that while you can see the different servers in the main right-click menu, the "Apps" menu does not indicate a server.

### Client configuration
Client version: 2.6.2

Operating system: win10_64

OS language: english

### Logs

<details><summary>nextcloud desktop log</summary>

```
[OCC::AccessManager::createRequest 	6 "PROPFIND" "https://nc.server1.com/cloud/remote.php/dav/files/myuser1/" has X-Request-ID "4338b1f1-5911-46bc-b45c-0b4b82a6af1b"
[OCC::AbstractNetworkJob::start 	OCC::PropfindJob created for "https://nc.server1.com/cloud" + "/" "OCC::QuotaInfo"
[OCC::WebFlowCredentials::slotFinished 	request finished
[OCC::PropfindJob::finished 	PROPFIND of QUrl("https://nc.server1.com/cloud/remote.php/dav/files/myuser1/") FINISHED WITH STATUS "OK"
[OCC::AccessManager::createRequest 	6 "GET" "https://nc.server1.com/cloud/ocs/v2.php/core/navigation/apps?absolute=true&format=json" has X-Request-ID "636f18cf-b7fe-445f-aad5-4dfa436053e0"
[OCC::AbstractNetworkJob::start 	OCC::OcsNavigationAppsJob created for "https://nc.server1.com/cloud" + "ocs/v2.php/core/navigation/apps" ""
[OCC::AccessManager::createRequest 	6 "GET" "https://nc.server2.com/oc/ocs/v2.php/core/navigation/apps?absolute=true&format=json" has X-Request-ID "276bd097-657f-42cb-9e8c-6abce2a709ac"
[OCC::AbstractNetworkJob::start 	OCC::OcsNavigationAppsJob created for "https://nc.server2.com/oc" + "ocs/v2.php/core/navigation/apps" ""
[OCC::WebFlowCredentials::slotFinished 	request finished
[OCC::OcsJob::finished 	Could not parse reply to "GET" QUrl("https://nc.server2.com/oc/ocs/v2.php/core/navigation/apps") (QPair("absolute","true")) "illegal value" : ""
[OCC::ownCloudGui::slotNavigationAppsFetched 	Status code  304  Not Modified - No new navigation apps.
[OCC::WebFlowCredentials::slotFinished 	request finished
[OCC::Folder::slotRunEtagJob 	Trying to check "https://nc.server1.com/cloud/remote.php/dav/files/myuser1/" for changes via ETag check. (time since last sync: 280 s)
[OCC::Folder::slotRunEtagJob 	Trying to check "https://nc.server2.com/oc/remote.php/dav/files/myuser2/" for changes via ETag check. (time since last sync: 628 s)
[OCC::AccessManager::createRequest 	6 "PROPFIND" "https://nc.server1.com/cloud/remote.php/dav/files/myuser1/" has X-Request-ID "c3dac906-f826-47a2-94e5-f7ab9515ac47"
[OCC::AbstractNetworkJob::start 	OCC::RequestEtagJob created for "https://nc.server1.com/cloud" + "/" "OCC::Folder"
[OCC::AccessManager::createRequest 	6 "GET" "https://nc.server1.com/cloud/ocs/v2.php/core/navigation/apps?absolute=true&format=json" has X-Request-ID "c8d45d6a-958d-4f35-a5fa-b7a07a12ce31"
[OCC::AbstractNetworkJob::start 	OCC::OcsNavigationAppsJob created for "https://nc.server1.com/cloud" + "ocs/v2.php/core/navigation/apps" ""
[OCC::AccessManager::createRequest 	6 "GET" "https://nc.server2.com/oc/ocs/v2.php/core/navigation/apps?absolute=true&format=json" has X-Request-ID "6e3d3210-7fbe-4eab-8a01-6fbb1fd141fe"
[OCC::AbstractNetworkJob::start 	OCC::OcsNavigationAppsJob created for "https://nc.server2.com/oc" + "ocs/v2.php/core/navigation/apps" ""
[OCC::WebFlowCredentials::slotFinished 	request finished
[OCC::OcsJob::finished 	Could not parse reply to "GET" QUrl("https://nc.server2.com/oc/ocs/v2.php/core/navigation/apps") (QPair("absolute","true")) "illegal value" : ""
[OCC::ownCloudGui::slotNavigationAppsFetched 	Status code  304  Not Modified - No new navigation apps.
[OCC::WebFlowCredentials::slotFinished 	request finished
[OCC::RequestEtagJob::finished 	Request Etag of QUrl("https://nc.server1.com/cloud/remote.php/dav/files/myuser1/") FINISHED WITH STATUS "OK"
[OCC::AccessManager::createRequest 	6 "PROPFIND" "https://nc.server2.com/oc/remote.php/dav/files/myuser2/" has X-Request-ID "4f215b6f-0735-4f56-b19d-ae98cce25ed1"
[OCC::AbstractNetworkJob::start 	OCC::RequestEtagJob created for "https://nc.server2.com/oc" + "/" "OCC::Folder"
[OCC::WebFlowCredentials::slotFinished 	request finished
[OCC::RequestEtagJob::finished 	Request Etag of QUrl("https://nc.server2.com/oc/remote.php/dav/files/myuser2/") FINISHED WITH STATUS "OK"
[OCC::WebFlowCredentials::slotFinished 	request finished
[OCC::AccessManager::createRequest 	6 "PROPFIND" "https://nc.server1.com/cloud/remote.php/dav/files/myuser1/" has X-Request-ID "e95c887a-5e1f-4960-ace1-bef464170e3b"
[OCC::AbstractNetworkJob::start 	OCC::PropfindJob created for "https://nc.server1.com/cloud" + "/" "OCC::ConnectionValidator"
[OCC::AccessManager::createRequest 	6 "PROPFIND" "https://nc.server2.com/oc/remote.php/dav/files/myuser2/" has X-Request-ID "8dba5856-35f4-4ad2-9229-21c424775b24"
[OCC::AbstractNetworkJob::start 	OCC::PropfindJob created for "https://nc.server2.com/oc" + "/" "OCC::ConnectionValidator"
[OCC::WebFlowCredentials::slotFinished 	request finished
[OCC::PropfindJob::finished 	PROPFIND of QUrl("https://nc.server2.com/oc/remote.php/dav/files/myuser2/") FINISHED WITH STATUS "OK"
[OCC::WebFlowCredentials::slotFinished 	request finished
[OCC::PropfindJob::finished 	PROPFIND of QUrl("https://nc.server1.com/cloud/remote.php/dav/files/myuser1/") FINISHED WITH STATUS "OK"
[OCC::AccessManager::createRequest 	6 "GET" "https://nc.server1.com/cloud/ocs/v2.php/core/navigation/apps?absolute=true&format=json" has X-Request-ID "0cd4a048-f716-4ad1-be27-8b4ce2e4eb43"
[OCC::AbstractNetworkJob::start 	OCC::OcsNavigationAppsJob created for "https://nc.server1.com/cloud" + "ocs/v2.php/core/navigation/apps" ""
[OCC::AccessManager::createRequest 	6 "GET" "https://nc.server2.com/oc/ocs/v2.php/core/navigation/apps?absolute=true&format=json" has X-Request-ID "60d7c07b-8a74-41cf-8fb4-e9b89c69703a"
[OCC::AbstractNetworkJob::start 	OCC::OcsNavigationAppsJob created for "https://nc.server2.com/oc" + "ocs/v2.php/core/navigation/apps" ""
[OCC::WebFlowCredentials::slotFinished 	request finished
[OCC::OcsJob::finished 	Could not parse reply to "GET" QUrl("https://nc.server2.com/oc/ocs/v2.php/core/navigation/apps") (QPair("absolute","true")) "illegal value" : ""
[OCC::ownCloudGui::slotNavigationAppsFetched 	Status code  304  Not Modified - No new navigation apps.
[OCC::WebFlowCredentials::slotFinished 	request finished
[OCC::AccessManager::createRequest 	6 "PROPFIND" "https://nc.server1.com/cloud/remote.php/dav/files/myuser1/" has X-Request-ID "fa2ca8cf-22b2-4366-8345-1108a4716a7a"
[OCC::AbstractNetworkJob::start 	OCC::PropfindJob created for "https://nc.server1.com/cloud" + "/" "OCC::QuotaInfo"
[OCC::WebFlowCredentials::slotFinished 	request finished
[OCC::PropfindJob::finished 	PROPFIND of QUrl("https://nc.server1.com/cloud/remote.php/dav/files/myuser1/") FINISHED WITH STATUS "OK"
[OCC::Folder::slotRunEtagJob 	Trying to check "https://nc.server1.com/cloud/remote.php/dav/files/myuser1/" for changes via ETag check. (time since last sync: 310 s)
[OCC::Folder::slotRunEtagJob 	Trying to check "https://nc.server2.com/oc/remote.php/dav/files/myuser2/" for changes via ETag check. (time since last sync: 658 s)
[OCC::AccessManager::createRequest 	6 "PROPFIND" "https://nc.server1.com/cloud/remote.php/dav/files/myuser1/" has X-Request-ID "1237cd12-7c7d-48aa-8383-5fc7da1f11b9"
[OCC::AbstractNetworkJob::start 	OCC::RequestEtagJob created for "https://nc.server1.com/cloud" + "/" "OCC::Folder"
[OCC::WebFlowCredentials::slotFinished 	request finished
[OCC::RequestEtagJob::finished 	Request Etag of QUrl("https://nc.server1.com/cloud/remote.php/dav/files/myuser1/") FINISHED WITH STATUS "OK"
[OCC::AccessManager::createRequest 	6 "PROPFIND" "https://nc.server2.com/oc/remote.php/dav/files/myuser2/" has X-Request-ID "cba1a369-8ba8-4239-8c30-861478150329"
[OCC::AbstractNetworkJob::start 	OCC::RequestEtagJob created for "https://nc.server2.com/oc" + "/" "OCC::Folder"
[OCC::WebFlowCredentials::slotFinished 	request finished
[OCC::RequestEtagJob::finished 	Request Etag of QUrl("https://nc.server2.com/oc/remote.php/dav/files/myuser2/") FINISHED WITH STATUS "OK"
[OCC::AccessManager::createRequest 	6 "PROPFIND" "https://nc.server1.com/cloud/remote.php/dav/files/myuser1/" has X-Request-ID "082a769d-0e87-4e41-8606-5d772b88d0eb"
[OCC::AbstractNetworkJob::start 	OCC::PropfindJob created for "https://nc.server1.com/cloud" + "/" "OCC::ConnectionValidator"
[OCC::AccessManager::createRequest 	6 "PROPFIND" "https://nc.server2.com/oc/remote.php/dav/files/myuser2/" has X-Request-ID "398ecd48-761c-4a27-986e-ed393ab95fdd"
[OCC::AbstractNetworkJob::start 	OCC::PropfindJob created for "https://nc.server2.com/oc" + "/" "OCC::ConnectionValidator"
[OCC::WebFlowCredentials::slotFinished 	request finished
[OCC::PropfindJob::finished 	PROPFIND of QUrl("https://nc.server2.com/oc/remote.php/dav/files/myuser2/") FINISHED WITH STATUS "OK"
[OCC::WebFlowCredentials::slotFinished 	request finished
[OCC::PropfindJob::finished 	PROPFIND of QUrl("https://nc.server1.com/cloud/remote.php/dav/files/myuser1/") FINISHED WITH STATUS "OK"
[OCC::AccessManager::createRequest 	6 "PROPFIND" "https://nc.server1.com/cloud/remote.php/dav/files/myuser1/" has X-Request-ID "b83d24fd-c705-43d5-a919-85e9993c5d97"
[OCC::AbstractNetworkJob::start 	OCC::PropfindJob created for "https://nc.server1.com/cloud" + "/" "OCC::QuotaInfo"
[OCC::WebFlowCredentials::slotFinished 	request finished
[OCC::PropfindJob::finished 	PROPFIND of QUrl("https://nc.server1.com/cloud/remote.php/dav/files/myuser1/") FINISHED WITH STATUS "OK"
[OCC::Folder::slotRunEtagJob 	Trying to check "https://nc.server1.com/cloud/remote.php/dav/files/myuser1/" for changes via ETag check. (time since last sync: 340 s)
[OCC::Folder::slotRunEtagJob 	Trying to check "https://nc.server2.com/oc/remote.php/dav/files/myuser2/" for changes via ETag check. (time since last sync: 688 s)
[OCC::AccessManager::createRequest 	6 "PROPFIND" "https://nc.server1.com/cloud/remote.php/dav/files/myuser1/" has X-Request-ID "b33e4020-cc60-4a4f-bc33-7046888f1375"
[OCC::AbstractNetworkJob::start 	OCC::RequestEtagJob created for "https://nc.server1.com/cloud" + "/" "OCC::Folder"
[OCC::WebFlowCredentials::slotFinished 	request finished
[OCC::RequestEtagJob::finished 	Request Etag of QUrl("https://nc.server1.com/cloud/remote.php/dav/files/myuser1/") FINISHED WITH STATUS "OK"
[OCC::AccessManager::createRequest 	6 "PROPFIND" "https://nc.server2.com/oc/remote.php/dav/files/myuser2/" has X-Request-ID "298c6858-a7a8-465e-ba2e-6ca68492e8eb"
[OCC::AbstractNetworkJob::start 	OCC::RequestEtagJob created for "https://nc.server2.com/oc" + "/" "OCC::Folder"
[OCC::WebFlowCredentials::slotFinished 	request finished
[OCC::RequestEtagJob::finished 	Request Etag of QUrl("https://nc.server2.com/oc/remote.php/dav/files/myuser2/") FINISHED WITH STATUS "OK"
```

</details>

Thank you.

**Comment by misch7** [comment]
This should be solved by our new UI in 2.7 where you have a separate Apps menu associated with the selected account.

Did you give it a try @r2evans?

The most recent version is 2.7.0 Beta 3 (RC1 coming soon):
https://github.com/nextcloud/desktop/releases/tag/v2.7.0-beta3

**Comment by r2evans** [comment]
Yes, I'm on beta3 and it appears this is no longer an issue. Thanks!
